pkg/exclusions/exclusions_test.go

```diff
@@ -0,0 +1,303 @@
+			wantErr:              true,
+		},
+	}
+	for _, tt := range tests {
+		tt := tt
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+			mockReporter := &MockReporter{}
+			lockfile := lockfile.Lockfile{Packages: tt.lockfileData}
+			filteredLockfile, err := ExcludePackages(mockReporter, tt.exclusionPatterns, &lockfile)
+
+			if !cmp.Equal(err != nil, tt.wantErr) {
+				t.Errorf("ExcludePackages returned an unexpected error: %v", err)
+			}
+
+			for i, pkg := range filteredLockfile.Packages {
+				if !cmp.Equal(pkg.Name, tt.expectedLockfileData[i].Name) {
+					t.Errorf("ExcludePackages did not filter the packages as expected. Got: %v, Want: %v", filteredLockfile.Packages, tt.expectedLockfileData)
+				}
+			}
+		})
+	}
+}
+
+func TestExcludeSBOMPackages(t *testing.T) {
+	t.Parallel()
+	type testCase struct {
+		name              string
+		SBOMData          sbom.Identifier
+		exclusionPatterns []string
+		wantMatch         bool
+		wantErr           bool
+	}
+
+	tests := []testCase{
+		{
+			name:              "Package matches regex pattern",
+			SBOMData:          sbom.Identifier{PURL: "package1"},
+			exclusionPatterns: []string{".*package.*"},
+			wantMatch:         true,
+			wantErr:           false,
+		},
+		{
+			name:              "No packages match regex pattern",
+			SBOMData:          sbom.Identifier{PURL: "package1"},
+			exclusionPatterns: []string{"somethingDifferent"},
+			wantMatch:         false,
+			wantErr:           false,
+		},
+		{
+			name:              "Invalid regex pattern",
+			SBOMData:          sbom.Identifier{PURL: "package1"},
+			exclusionPatterns: []string{"*hello*"},
+			wantMatch:         false,
+			wantErr:           true,
+		},
+	}
+	for _, tt := range tests {
+		tt := tt
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+			mockReporter := &MockReporter{}
+			match, err := ExcludeSBOMPackages(mockReporter, tt.exclusionPatterns, &tt.SBOMData)
+
+			if !cmp.Equal(match, tt.wantMatch) {
+				t.Errorf("Expected match: %v, got match: %v", tt.wantMatch, match)
+			}
+
+			if !cmp.Equal(err != nil, tt.wantErr) {
+				t.Errorf("Expected error: %v, got error: %v", tt.wantErr, err)
+			}
+		})
+	}
+}
+
+func TestIsRegexPatterns(t *testing.T) {
+	t.Parallel()
+	type testCase struct {
+		name    string
+		pattern []string
+		want    bool
+	}
+	testCases := []testCase{
+		{
+			name:    "Single valid regex pattern",
+			pattern: []string{".*aValidPattern"},
+			want:    false,
+		},
+		{
+			name:    "Multiple valid regex patterns",
+			pattern: []string{".*aValidPattern", "another_valid_{r1}egex"},
+			want:    false,
+		},
+		{
+			name:    "Invalid regex pattern",
+			pattern: []string{"*myInvalidPattern*["},
+			want:    true,
+		},
+	}
+
+	for _, tt := range testCases {
+		tt := tt
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+			err := IsRegexPatterns(tt.pattern)
+			if (err != nil) != tt.want {
```

Comment:
This would be much better with some assertion library. Not sure of a better way to write this with the standard libs.